### ISSUE TITLE
Documentation for ZFS_POOL_IMPORT is slightly misleading

### ISSUE THREAD

**Issue by yacoob** [zfsonlinux/zfs#9126]
The comment in `/etc/default/zfs` says:
```
# List of pools that SHOULD be imported at boot by the initramfs
# instead of trying to import all available pools.  If this is set
# then ZFS_POOL_EXCEPTIONS is ignored.
# Only applicable for Debian GNU/Linux {dkms,initramfs}.
# This is a semi-colon separated list.
#ZFS_POOL_IMPORT="pool1;pool2"
```
This suggests that if you set this, the pools will always be imported. What actually happens is that `ZFS_POOL_IMPORT` is only consulted if you set `root=zfs:AUTO` and the initramfs script needs to work out the name of the pool containing the root device on its own.

I thought I can use it to just import arbitrary pools at the boot time. :)

**Comment by kpfleming** [comment]
Agreed, I also attempted to use this setting to import additional pools and was surprised when it failed.

**Comment by mcdonc** [comment]
It appears this feature is now vestigial

Introduced in https://github.com/openzfs/zfs/commit/57732964d35616d6c93b8377b1ed72c485d9d73d#diff-98a21a3fd74b681e806ecbd958c2352b 

There *was* a file named named "contrib/initramfs/scripts/zfs.in" that referenced it, but it was renamed to "zfs", and I *think* the commit that made the feature vestigial is the same one that did the rename... https://github.com/openzfs/zfs/commit/01243e72a5d7144345a9c13ab4455a52003fc0af#diff-98a21a3fd74b681e806ecbd958c2352b

I'm not sure whether the feature should be restored or if the lines in etc/default/zfs.in at https://github.com/openzfs/zfs/blob/master/etc/default/zfs.in#L55-L60 should just be removed and a news entry added about the feature removal?

**Comment by mcdonc** [comment]
This may need input from @behlendorf and/or @rlaager

**Comment by behlendorf** [comment]
Unless there's a compelling use case to restore this functionality I'd suggest we remove it.

**Comment by mcdonc** [comment]
There are some other features in the same state.  I audited each envvar in the etc/default/zfs.in file, and removed references to these features:

ZFS_POOL_IMPORT
ZFS_INITRD_PRE_MOUNTROOT_SLEEP
ZFS_INITRD_POST_MODPROBE_SLEEP
ZFS_INITRD_ADDITIONAL_DATASETS

I created a PR at https://github.com/openzfs/zfs/pull/10757

**Comment by mcdonc** [comment]
A side note, I opened a new issue at #10756 about keeping track of feature changes in a repo file.  This particular change may be in the weeds, so a description of it may not be a candidate for that file given the "real" features between 0.8 and 2.0, but I suspect someone who depends on these features would be grateful if some sort of detailed change notes had stuff like this in it, so maybe.
